### ISSUE TITLE
[New Pattern] Duplicate array keys

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1190]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | New Pattern                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.12 |
| Language level | PHP 7.3                        |

Current behaviour:
---

Here is some example code:

```php
<?php

declare(strict_types=1);

final class FooBar
{
    public function getString(): string
    {
        return 'string';
    }

    public function getRandomString(): string
    {
        return (string)random_int(1, 999999);
    }
}

$fooBar = new FooBar();

$test = [
    $fooBar->getString() => 123,
    $fooBar->getString() => 456
];

$testRandom = [
    $fooBar->getRandomString() => 123,
    $fooBar->getRandomString() => 456
];
```

No warning is displayed.

Suggested behaviour:
---

Display a warning that the key is (most probably) duplicated. Sure, obviously it could also return something randomly like in the second example with the call to `getRandomString()`, but I would consider that dirty and worth to be reported as well, although I'm not absolutely sure myself about that behaviour.

**Comment by funivan** [comment]
> but I would consider that dirty and worth to be reported as well,

What to do with such errors: suppress, leave, refactor code?

**Comment by orklah** [comment]
We should at least exclude every method called on Iterator classes ->getNext() would be a perfectly valid call

**Comment by andreasschroth** [comment]
@orklah Yep, true, 100% agreed! Didn't think about a specific call on that method.

**Comment by andreasschroth** [comment]
> > but I would consider that dirty and worth to be reported as well,
> 
> What to do with such errors: suppress, leave, refactor code?

@funivan Good question, actually I can't even think of a valid use-case (besides an Iterator as mentioned by @orklah ). It might actually even be a bug. But if it's valid code, then I guess a suppression.

**Comment by andreasschroth** [comment]
Closing this as I don't think it's a good proposal to implement.
